cl/citations/filter_parentheticals.py

```diff
@@ -0,0 +1,76 @@
+import re
```

Comment:
Two brought thoughts on this file:

1. Regexes are just the worst. Too bad they're also the best. Given that, would it be possible to take a run at refactoring this file to make it even cleaner than it is already? I've highlighted a couple things so far.

2. I think another thing that'd help — I think — would be to eliminate the non-capturing groups. Unless I'm mistaken capturing or not is fine? That'd eliminate a bunch of question marks.

Reply:
Sorry, one more: 

3. Are all of these tested? I see a bunch of tests, but haven't checked if everything here has a test. With something like this, a big test suite is going to be essential.

Reply:
1. https://xkcd.com/1171/ Yep, I'll make those changes and see what else I can do to clean things up.
2. That's a good point, I'll do that.
3. Many but definitely not all of the different cases are tested. We will add some more test cases to make it a bit more comprehensive.